homeassistant/components/vesync/__init__.py

```diff
@@ -48,6 +51,25 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
     hass.data[DOMAIN] = {}
     hass.data[DOMAIN][VS_MANAGER] = manager
 
+    # Create a DataUpdateCoordinator for the manager
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        try:
+            await hass.async_add_executor_job(manager.update)
+        except Exception as err:
+            raise UpdateFailed(f"Update failed: {err}") from err
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=DOMAIN,
+        update_method=async_update_data,
+        update_interval=timedelta(seconds=30),
+    )
```

Comment:
Let's make this a separate class in `coordinator.py`. A lot of other integrations have a `coordinator.py` and are good examples for this